helper/resource/error.go

```diff
@@ -0,0 +1,21 @@
+package resource
+
+type NotFoundError struct {
+	LastError    error
+	LastRequest  interface{}
+	LastResponse interface{}
+	Message      string
+	Retries      int
+}
+
+func (e *NotFoundError) Error() string {
+	if e.Message != "" {
+		return e.Message
+	}
+
+	return "couldn't find resource"
+}
+
+func NewNotFoundError(err string) *NotFoundError {
```

Comment:
I'm actually wondering if there is actually any value in those wrapper functions and whether we shouldn't just remove this function completely and leave the implementation like this: `&NotFoundError{Param: value}`.

If you think about cases where you'll need to define more parameters, which on the resource-level is often, this would IMO become slightly messier:

```go
return resource.NotFoundError(err).WithMessage("Lambda Permissions couldn't be found").WithLastRequest(req).WithLastResponse(resp).WithRetries(retries)
```
compared to

```go
return &NotFoundError{
    Message: "Lambda Permissions couldn't be found",
    LastRequest: req,
    LastResponse: resp,
    Retries: retries,
}
```


:thought_balloon: